src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -268,6 +268,10 @@ public function lists($column, $key = null)
      */
     public function paginate($perPage = null, $columns = ['*'], $pageName = 'page', $page = null)
     {
+        if ($perPage < 0) {
+            throw new \Exception("Negative values can't be used to query a result set");
```

Comment:
Please throw an invalid argument exception, and import it instead of using a slash.